### PR TITLE
A (hopefully) better error message

### PR DESCRIPTION
**Pull request by kmclarnon**
This error message caused me a little bit of confusion when I was trying to help someone who was running into this issue.  I read project and though the entire repo, rather than the module being built, and the appropriate resolution wasn't immediately clear to me (even though after the fact this makes perfect sense)

@stevegutz @snommit-mit @Xcelled

### CODE REVIEW

**Comment by jhaber** [comment]
lgtm 🚢

File: src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java [hunk 1]
```diff
@@ -61,8 +61,10 @@ private boolean checkDependencyManagement() {
         String managedVersion = managedDependency.getVersion();
 
         if (!projectVersion.equals(managedVersion)) {
-          String errorFormat = "Version mismatch for %s, managed version %s does not match project version %s";
+          String errorFormat = "Version mismatch for %s, managed version %s does not match version %s declared in module's pom";
+          String additionalMessage = "If you intended to use the version specified in dependency management, please remove the version from this module's dependency declaration.";
```
Comment (jhaber) [inline src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java:65]: There's also the inverse, which is if you intended to use the local version you should hoist it to dep management, but not sure if we want to suggest that

Comment (kmclarnon) [reply src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java:65]: Ah hmm I was thinking that in that case they'd trip the `Found unmanaged dependencies` message.

Comment (jhaber) [reply src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java:65]: I meant in the case where managed version != project version, there's basically two options to resolve. One option is to remove your local version, but another option is to move your local version into a dependency management section. Currently we're only suggesting the first option, which is maybe desirable?

Comment (kmclarnon) [reply src/main/java/com/hubspot/maven/plugins/dependency/management/DependencyManagementAnalyzer.java:65]: Yea I think so, I don't think we really want to encourage adding things to local dependency management 👍

**Pull request merged**